teamsupport/services.py

```diff
@@ -40,16 +40,46 @@ def __init__(self, org_id, auth_token, **kwargs):
         self.org_id = org_id
         self.auth_token = auth_token
 
-    def get_tickets(self, query_params=None):
+    def search_tickets(self, query_params=None):
         response = self.get('Tickets/', params=query_params)
         content = self.parse_xml_response(response)
         return content
 
+    def search_contacts(self, **query_params):
+        response = self.get('Contacts/', params=query_params)
+        return self.parse_xml_response(response)
+
+    def create_contact(self, **data):
+        response = self.post(
+            'Contacts/', root='Contact', data=data, send_as_xml=True)
+        return self.parse_xml_response(response)
+
+    def delete_contact(self, contact_id):
+        self.delete('Contacts/{}'.format(contact_id))
+
     def create_ticket(self, data):
         response = self.post(
-            'Tickets/', root='Ticket', data=data, send_as_xml=True)
+            'tickets', root='Ticket', data=data, send_as_xml=True)
         return self.parse_xml_response(response)
 
+    def set_ticket_description(self, ticket_id, description):
+        # Description is an Action in TeamSupport API. That action is created
+        # automatically when the ticket is created. We need to query it's ID
+        # and update this action to set ticket description.
+        ticket_actions = self.get_ticket_actions(
+            ticket_id, {'SystemActionTypeID': 1})
+        action_id = ticket_actions[0].find('ActionID').text
+        self.update_ticket_action(
+            ticket_id, action_id, {'Description': description})
+
+    def get_ticket_description(self, ticket_id):
```

Comment:
I consider services and models as different levels of same API. If you need more flexibility and power, you can use services. If you want more readability and simplicity - you can use models. I disagree that service methods must always correspond API calls. Why? Also, this case is a quite specific one. It is clearly specified in TeamSupport API docs (totally incomplete btw) that the description is an action with certain ID. So, this is an API level, isn't it?